cpp/benchmarks/streaming/ndsh/q17.cpp

```diff
@@ -0,0 +1,1195 @@
+) {
+    rapidsmpf::streaming::ShutdownAtExit c{ch_in, ch_out};
+    while (true) {
+        auto msg = co_await ch_in->receive();
+        if (msg.empty()) {
+            break;
+        }
+        co_await ctx->executor()->schedule();
+        auto chunk = rapidsmpf::ndsh::to_device(
+            ctx, msg.release<rapidsmpf::streaming::TableChunk>()
+        );
+        auto chunk_stream = chunk.stream();
+        auto sequence_number = msg.sequence_number();
+        auto table = chunk.table_view();
+
+        std::vector<std::unique_ptr<cudf::column>> result;
+        result.reserve(indices.size());
+        for (auto idx : indices) {
+            result.push_back(
+                std::make_unique<cudf::column>(
+                    table.column(idx), chunk_stream, ctx->br()->device_mr()
+                )
+            );
+        }
+
+        auto result_table = std::make_unique<cudf::table>(std::move(result));
+        co_await ch_out->send(
+            rapidsmpf::streaming::to_message(
+                sequence_number,
+                std::make_unique<rapidsmpf::streaming::TableChunk>(
+                    std::move(result_table), chunk_stream
+                )
+            )
+        );
+    }
+    co_await ch_out->drain(ctx->executor());
+}
+
+// Node to compute sum and count of quantity per partkey
+rapidsmpf::streaming::Node compute_avg_quantity(
+    std::shared_ptr<rapidsmpf::streaming::Context> ctx,
+    std::shared_ptr<rapidsmpf::streaming::Channel> ch_in,
+    std::shared_ptr<rapidsmpf::streaming::Channel> ch_out
+) {
+    rapidsmpf::streaming::ShutdownAtExit c{ch_in, ch_out};
+    std::uint64_t sequence = 0;
+    while (true) {
+        auto msg = co_await ch_in->receive();
+        if (msg.empty()) {
+            break;
+        }
+        co_await ctx->executor()->schedule();
+        auto chunk = rapidsmpf::ndsh::to_device(
+            ctx, msg.release<rapidsmpf::streaming::TableChunk>()
+        );
+        auto chunk_stream = chunk.stream();
+        auto table = chunk.table_view();
+
+        // table has: p_partkey, l_quantity, l_extendedprice
+        // Group by p_partkey and compute sum and count of l_quantity
+        auto grouper = cudf::groupby::groupby(
+            table.select({0}), cudf::null_policy::EXCLUDE, cudf::sorted::NO
+        );
+        auto requests = std::vector<cudf::groupby::aggregation_request>();
+        std::vector<std::unique_ptr<cudf::groupby_aggregation>> sum_aggs;
+        sum_aggs.push_back(cudf::make_sum_aggregation<cudf::groupby_aggregation>());
+        std::vector<std::unique_ptr<cudf::groupby_aggregation>> count_aggs;
+        count_aggs.push_back(cudf::make_count_aggregation<cudf::groupby_aggregation>());
+        requests.push_back(
+            cudf::groupby::aggregation_request(table.column(1), std::move(sum_aggs))
+        );
+        requests.push_back(
+            cudf::groupby::aggregation_request(table.column(1), std::move(count_aggs))
+        );
```

Comment:
I think this could be simplified as follows, since both sum and count are happening on col 1. 
```suggestion
        std::vector<std::unique_ptr<cudf::groupby_aggregation>> aggs;
        aggs.push_back(cudf::make_sum_aggregation<cudf::groupby_aggregation>());
        aggs.push_back(cudf::make_count_aggregation<cudf::groupby_aggregation>());
        requests.push_back(
            cudf::groupby::aggregation_request(table.column(1), std::move(aggs))
        );
```